src/status_im2/contexts/onboarding/common/overlay/view.cljs

```diff
@@ -0,0 +1,67 @@
+(ns status-im2.contexts.onboarding.common.overlay.view
+  (:require [react-native.reanimated :as reanimated]
+            [react-native.blur :as blur]
+            [react-native.core :as rn]
+            [reagent.core :as reagent]
+            [status-im2.contexts.onboarding.common.overlay.style :as style]
+            [status-im2.constants :as constants]))
+
+(def max-blur-amount 30)
+
+(defonce timer-interval (atom nil))
+(defonce blur-dismiss-fn-atom (atom nil))
+(defonce blur-show-fn-atom (atom nil))
+
+(defn f-view
+  [blur-amount]
+  (let [opacity                     (reanimated/use-shared-value 0)
+        transition-half-duration-ms (/ constants/onboarding-modal-animation-duration 2)
+        blur-show-fn                (fn []
+                                      (reanimated/animate opacity
+                                                          1
+                                                          transition-half-duration-ms)
+                                      (js/clearInterval @timer-interval)
+                                      (reset! timer-interval
+                                        (js/setInterval
+                                         (fn []
+                                           (if (< @blur-amount max-blur-amount)
+                                             (swap! blur-amount + 1)
+                                             (js/clearInterval @timer-interval)))
+                                         (/ constants/onboarding-modal-animation-duration
+                                            max-blur-amount
+                                            2))))
+        blur-dismiss-fn             (fn []
+                                      (reanimated/animate-delay opacity
+                                                                0
+                                                                transition-half-duration-ms
+                                                                transition-half-duration-ms)
+                                      (reset! timer-interval
+                                        (js/setInterval
+                                         (fn []
+                                           (if (> @blur-amount 0)
+                                             (swap! blur-amount - 1)
+                                             (js/clearInterval @timer-interval)))
+                                         (/ constants/onboarding-modal-animation-duration
+                                            max-blur-amount
+                                            2))))]
+    (rn/use-effect
```

Comment:
consider using `reagent/with-let`. cc @ulisesmac

Reply:
@briansztamfater  @J-Son89 
 
 I thought the same at first sight, then I tried to modify the code to achieve it but it's not possible.
 
The functions @briansztamfater is passing to the atoms depend on a reanimated shared value, which is a hook. so, the problem is, the bindings in `with-let` are executed only once (when the component is mounted) and we'd need to create the `opacity`  shared value there, but in React  we must use the same amount of hooks in every re-render.

btw, I haven't shared the with-let explanation, if you want to read it:
https://github.com/reagent-project/reagent/blob/master/doc/CreatingReagentComponents.md#appendix-b---with-let-macro